### PR TITLE
Root and SignedRoot

### PR DESCRIPTION
**Pull request by jannikluhn**
## What was wrong?

#44 

## How was it fixed?

- add the property `root` to `BaseSerializable`
- add a metaclass for signed serializables that validates that the last field is called signature and there's at least one more (otherwise we can't calculate the signed root, assuming empty containers don't exist)
- add a new base class `SignedSerializable` that provides the property `signed_root`

(builds on top of #57)

#### Cute Animal Picture

put a cute animal picture link inside the parentheses

### CODE REVIEW

File: ssz/sedes/serializable.py [hunk 1]
```diff
@@ -27,11 +35,13 @@
 TSerializable = TypeVar("TSerializable", bound="BaseSerializable")
 
 
-class MetaBase:
-    fields = None
-    field_names = None
-    field_attrs = None
-    container_sedes = None
+class Meta(NamedTuple):
+
```
Comment (ralexstokes) [inline ssz/sedes/serializable.py]: minor but i feel like we usually have no whitespace b/t class decl and class attrs

File: ssz/sedes/signed_serializable.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    Container,
+)
+from ssz.sedes.serializable import (
+    BaseSerializable,
+    MetaSerializable,
+)
+
+
+class SignedMeta(NamedTuple):
+    has_fields: bool
```
Comment (ralexstokes) [inline ssz/sedes/signed_serializable.py]: example about formatting here

File: ssz/sedes/signed_serializable.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    fields: Optional[Tuple[Tuple[str, BaseSedes]]]
+    container_sedes: Optional[Container]
+    signed_container_sedes: Optional[Container]
+    field_names: Optional[Tuple[str]]
+    field_attrs: Optional[Tuple[str]]
+
+
+class MetaSignedSerializable(MetaSerializable):
+
```
Comment (ralexstokes) [inline ssz/sedes/signed_serializable.py]: whitespace? i'll let you decide if you want to patch these up, not blocking merging imo

File: ssz/sedes/serializable.py (hunk 1 above)
Comment (jannikluhn) [reply ssz/sedes/serializable.py]: :+1: I don't like how it looks, but I do like conventions, so I removed the new lines for all classes I could find.

**Pull request merged**